### DISCUSSION TITLE
Are action atoms still recommended?

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1449] [Q&A] [answered]
I was curious if action atoms still should be used.

See p. 163 "using action atoms".
https://www.amazon.com/Micro-State-Management-React-Hooks-dp-1801812373/dp/1801812373/

Looks like they easily can be replaced by calling `useAtomCallback`

**Comment by dai-shi** [comment, marked as answer]
`useAtomCallback` is theoretically heavier than action atoms (as it's implemented with action atoms). I'd avoid using `useAtomCallback` unless it's really necessary (one benefit is it hacks to return a promise value). I prefer action atoms that allows us to define logic outside components and independent from React (yes, atoms are not tied to React.)
